### DISCUSSION TITLE
Question about Lynns Airplane Example in the Playbook

### DISCUSSION THREAD

**Discussion by QtheFox** [leap71/PicoGK#46] [Engineering Discussion]
Hello,
with great interest I have read everything about picogk and tried it myself.
I really appreciated the example for alogirthmic engineering of an airplane.
Now I am wondering how you could define boundary conditions and relations between 2 dependent parts. And how an algorithmic solution would evolve. Could you give in pseudocode a simple example for this? like a sketch..
for example:
turbofan engine with the parameters: thrust, weight, size, cog
wing with parameters: profile, length, no aspect ratio, volume for fuel, cog, weight
fuselage with parameters: number of seats, diameter, air resistance, cog, weight

now I want an airplane optimized to 800 km/h speed and 50 seats

Would be very helpful.

**Comment by LinKayser** [comment]
This will be part of a significant portion of the book. The short answer is: You need to communicate between the different "parts", but the "part" may not be a physical part — it's more a logical structure. The physical manifestation of the object doesn't necessarily map 1:1 to the logical hierarchy.

Generally speaking, you are trying to communicate all the physical constraints, the engineering logic, etc. between the two parts. The geometry is then the result of an, often iterative, negotiation between the parts.

But an airplane is a very complex object. So you will be running through an enormously deep hierarchy of dependencies. The good thing is, that at the bottom layer, these dependencies are relatively trivial, so you can work yourself up from these smaller entities.

Lin
